SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp

```diff
@@ -226,21 +229,75 @@ void USpatialGDKEditorSettings::SetNumberOfSimulatedPlayers(uint32 Number)
 	SaveConfig();
 }
 
-bool USpatialGDKEditorSettings::IsManualWorkerConnectionSet(const FString& LaunchConfigPath)
+bool USpatialGDKEditorSettings::IsManualWorkerConnectionSet(const FString& LaunchConfigPath, TArray<FString>& OutWorkersManuallyLaunched)
 {
-	FString FileContents;
-	FFileHelper::LoadFileToString(FileContents, *LaunchConfigPath);
+	TSharedPtr<FJsonValue> LaunchConfigJson;
+	{
+		FArchive* ConfigFile = IFileManager::Get().CreateFileReader(*LaunchConfigPath);
+
+		if (!ConfigFile)
+		{
+			UE_LOG(LogSpatialEditorSettings, Error, TEXT("Could not open configuration file %s"), *LaunchConfigPath);
+			return false;
+		}
+
+		TSharedRef<TJsonReader<char>> JsonReader = TJsonReader<char>::Create(ConfigFile);
+
+		FJsonSerializer::Deserialize(*JsonReader, LaunchConfigJson);
+
+		delete ConfigFile;
+		ConfigFile = nullptr;
+	}
+
+	const TSharedPtr<FJsonObject>* LaunchConfigJsonRootObject;
+
+	if (!LaunchConfigJson || !LaunchConfigJson->TryGetObject(LaunchConfigJsonRootObject))
+	{
+		UE_LOG(LogSpatialEditorSettings, Error, TEXT("Invalid configuration file %s"), *LaunchConfigPath);
+		return false;
+	}
+
+	const TSharedPtr<FJsonObject>* LoadBalancingField;
+	if (!(*LaunchConfigJsonRootObject)->TryGetObjectField("load_balancing", LoadBalancingField))
+	{
+		return false;
```

Comment:
can we log some error here as well to indicate where the problem is?

Reply:
Is it an error to not have a load_balancing section ?

Reply:
No it's not an error